addons/repair/models/account_move.py

```diff
@@ -8,6 +8,10 @@ class AccountMove(models.Model):
 
     repair_ids = fields.One2many('repair.order', 'invoice_id', readonly=True, copy=False)
 
+    def unlink(self):
+        self.repair_ids.state = '2binvoiced'
```

Comment:
self.repair_ids.filtered(lambda repair: repair.state != 'cancel').state = '2binvoiced'

else if you unlink the canceled draft of a cancelled repair, it goes to the 2binvoiced state